docs/specs/metadata.yml

```diff
@@ -125,3 +125,96 @@ outputs:
   azure/billing/billing-add-change-azure-subscription-administrator.json: |
     {"id":"f5624f3e-5fbf-6f0f-93cf-a4d5aa93ccf8","versionIndependentId":"9ca6d22b-e684-4e65-cab3-c272189a3ddc"}
   build.manifest:
+---
+# yaml&json toc metadata
+inputs:
+  docfx.yml:
+  docs/json/TOC.json: |
+    {
+      "metadata": {"a":"b"},
+      "items": [{ "name": "title" }]
+    }
+  docs/yaml/TOC.yml: |
+    metadata:
+      a: b
+    items:
+      - name: title
+outputs:
+  docs/json/TOC.json: |
+    {  
+      "metadata": {"a": "b"},
+      "items":[{"toc_title":"title"}]
+    }
+  docs/yaml/TOC.json: |
+    {  
+      "metadata": {"a": "b"},
+      "items":[{"toc_title":"title"}]
+    }
+  build.manifest:
+---
+# yaml&json toc leaf metadata
+inputs:
+  docfx.yml:
+  docs/json/TOC.json: |
+    [{ "name": "title", "a": "b" }]
+  docs/yaml/TOC.yml: |
+    - name: title
+      a: b
+outputs:
+  docs/json/TOC.json: |
+    {  
+      "items":[{"toc_title":"title", "a": "b"}]
+    }
+  docs/yaml/TOC.json: |
+    {  
+      "items":[{"toc_title":"title", "a": "b"}]
+    }
+  build.manifest:
+---
+# yaml&json toc metadata inclusion
+# child's root metadata will be dropped
+inputs:
+  docfx.yml:
+  docs/json/TOC.json: |
+    {  
+      "metadata": {"a": "b"},
+      "items":[{"name":"title", "c": "d"}]
+    }
+  docs/yaml/TOC.yml: |
+    - name: toc reference
+      tocHref: ../json/TOC.json
+outputs:
+  docs/yaml/TOC.json: |
+    {  
+      "items":[{"toc_title":"toc reference", "children": [{"toc_title":"title", "c": "d"}]}]
```

Comment:
> "c": "d" [](start = 79, length = 9)

> # child's root metadata will be dropped
It's not dropped. conflict with test title?

Reply:
As discussion, ~~child~~**embeded toc**'s root metadata will be dropped

---
In reply to: [206760980](https://github.com/dotnet/docfx/pull/3141#discussion_r206760980) [](ancestors = 206760980)

Reply:
O

---
In reply to: [206762775](https://github.com/dotnet/docfx/pull/3141#discussion_r206762775) [](ancestors = 206762775,206760980)